src/main/java/com/meta/cp4m/message/FBMessengerConfig.java

```diff
@@ -10,60 +10,77 @@
-    return name;
-  }
-
-  public String verifyToken() {
-    return verifyToken;
-  }
-
-  public String appSecret() {
-    return appSecret;
-  }
-
-  @Override
-  public FBMessageHandler toMessageHandler() {
-    return new FBMessageHandler(this);
-  }
-
-  public String pageAccessToken() {
-    return pageAccessToken;
-  }
-}
+    private final String name;
+    private final String verifyToken;
+    private final String appSecret;
+    private final String pageAccessToken;
+    @Nullable
+    private final String connectedFacebookPageForInstagram;
+
+    private FBMessengerConfig(
+            @JsonProperty("name") String name,
+            @JsonProperty("verify_token") String verifyToken,
+            @JsonProperty("app_secret") String appSecret,
+            @JsonProperty("page_access_token") String pageAccessToken,
+            @JsonProperty("connected_facebook_page_for_instagram") @Nullable String connectedFacebookPageForInstagram) {
+
+        Preconditions.checkArgument(name != null && !name.isBlank(), "name cannot be blank");
+        Preconditions.checkArgument(
+                verifyToken != null && !verifyToken.isBlank(), "verify_token cannot be blank");
+        Preconditions.checkArgument(
+                appSecret != null && !appSecret.isBlank(), "app_secret cannot be blank");
+        Preconditions.checkArgument(
+                pageAccessToken != null && !pageAccessToken.isBlank(), "page_access_token cannot be blank");
+
+        this.name = name;
+        this.verifyToken = verifyToken;
+        this.appSecret = appSecret;
+        this.pageAccessToken = pageAccessToken;
+        this.connectedFacebookPageForInstagram = connectedFacebookPageForInstagram;
+    }
+
+    public static FBMessengerConfig of(String verifyToken, String appSecret, String pageAccessToken, @Nullable String connectedFacebookPageForInstagram) {
+        // human readability of the name only matters when it's coming from a config
+        return new FBMessengerConfig(
+                UUID.randomUUID().toString(), verifyToken, appSecret, pageAccessToken, connectedFacebookPageForInstagram);
+    }
+
+    public static FBMessengerConfig of(String verifyToken, String appSecret, String pageAccessToken) {
+        // human readability of the name only matters when it's coming from a config
+        return new FBMessengerConfig(
+                UUID.randomUUID().toString(), verifyToken, appSecret, pageAccessToken, null);
+    }
+
+    @Override
+    public String name() {
+        return name;
+    }
+
+    public String verifyToken() {
+        return verifyToken;
+    }
+
+    public String appSecret() {
+        return appSecret;
+    }
+
+    @Override
+    public FBMessageHandler toMessageHandler() {
+        return new FBMessageHandler(this);
+    }
+
+    public String pageAccessToken() {
+        return pageAccessToken;
+    }
+
+    public String connectedFacebookPageForInstagram() {
+        return connectedFacebookPageForInstagram;
```

Comment:
```suggestion
    public Optional<String> connectedFacebookPageForInstagram() {
        return Optional.ofNullable(connectedFacebookPageForInstagram);
```

pretty sure this the right code, but you want a return an Optional where possible for nullable values from functions